lib/whitehall/asset_manager_storage.rb

```diff
@@ -11,12 +11,23 @@ def store!(file)
     FileUtils.cp(original_file, temporary_location)
     legacy_url_path = ::File.join("/government/uploads", uploader.store_path)
     draft = uploader.assets_protected?
+
+    model_class, model_id = get_model_class_and_model_id
+    is_attachable = model_class && uploader.model.respond_to?(:attachable)
+
+    AssetManagerCreateWhitehallAssetWorker.perform_async(temporary_location, legacy_url_path, draft, model_class, model_id, is_attachable)
+    File.new(uploader.store_path)
+  end
+
+  def get_model_class_and_model_id
     if uploader.model.respond_to?(:attachable)
       model_class = uploader.model.attachable && uploader.model.attachable.class.to_s
       model_id = uploader.model.attachable && uploader.model.attachable.id
+    elsif uploader.model.respond_to?(:images)
+      model_class = uploader.model.class.to_s
+      model_id = uploader.model.id
     end
```

Comment:
I think we can tidy up this implementation.

Let's make the worker responsible for this bit instead. So this class doesn't behave differently depending on `:attachable` behaviour.

Reply:
Resolving because we've gone for a different approach now